server/src/main/java/io/druid/client/HttpServerInventoryView.java

```diff
@@ -625,32 +605,80 @@ public void onFailure(Throwable t)
             },
             executor
         );
-
-        return future;
       }
       catch (Throwable th) {
-        addNextSyncToWorkQueue(druidServer.getName());
+        try {
+          String logMsg = StringUtils.nonStrictFormat(
+              "Fatal error while fetching segment list from server [%s].", druidServer.getName()
+          );
 
-        String logMsg = StringUtils.nonStrictFormat(
-            "Fatal error while fetching segment list from server [%s].", druidServer.getName()
-        );
+          if (hasUnstabilityTimeoutPassed()) {
+            log.makeAlert(th, logMsg).emit();
+          } else {
+            log.info("Temporary Failure. %s", logMsg);
+            log.debug(th, logMsg);
+          }
 
-        if (hasUnstabilityTimeoutPassed()) {
-          log.makeAlert(th, logMsg).emit();
-        } else {
-          log.info("Temporary Failure. %s", logMsg);
-          log.debug(th, logMsg);
+          // sleep for a bit so that retry does not happen immediately.
+          try {
+            Thread.sleep(5000);
```

Comment:
I guess RetryUtils.awaitNextRetry() is also useful here which has a sort of exponential backoff mechanism.

Reply:
Same for https://github.com/druid-io/druid/pull/4767/files#diff-dae68c8f716e1fd73cd6b671a5864fe1R595

Reply:
yeah, but i have been testing this class for a while on our server and haven't seen the need for exponential backoff. i will add that if it becomes necessary. (trying to limit scope of this pr)

Reply:
I wonder why you think its need is not so big. Sleeping 5 seconds here means that this server's segment information will be updated after at least 5 seconds. This should be decreased if possible while avoiding making a huge overhead to the entire system. 

I'm fine with opening a new issue for this if you think this is not in the scope of this pr, but I think it's definitely needed.

Reply:
ok, thinking more about it:
worst case scenario: none of the data nodes are reachable and this is trying to send a request every 5 seconds (assuming end servers are not blocking the request or else retry doesn't happen till request timeout). then, httpClient would be asked to do xx (based on cluster size) number of request every 5 seconds which almost immediately fail, this will likely increase cpu utilization on machine.
on the plus side we are trying every 5 seconds and segment list generally wouldn't get out of sync for more than that time even if a server failed a couple of times.

I think doing exponentially backoff is not necessarily bad, so I will add that in this PR instead of leaving it for future.

Reply:
ok, updated to do exponential backoff.

Reply:
Thanks.